Sources/ArgumentParser/Parsable Properties/NameSpecification.swift

```diff
@@ -38,10 +38,16 @@ public struct NameSpecification: ExpressibleByArrayLiteral {
     /// Short labels can be combined into groups.
     case customShort(Character)
   }
-  var elements: Set<Element>
+  var elements: Array<Element>
   
   public init<S>(_ sequence: S) where S : Sequence, Element == S.Element {
-    self.elements = Set(sequence)
+    let array = Array(sequence)
+    self.elements = Array()
+    for element in array {
+      if !self.elements.contains(element) {
+        self.elements.append(element)
+      }
+    }
```

Comment:
I feel like this would be easier to understand and also less complex if it were written like this:

```swift
  public init<S>(_ sequence: S) where S : Sequence, Element == S.Element {
    self.elements = []
    for element in Array(sequence) where !self.elements.contains(element) {
      self.elements.append(element)
    }
  }
```

Reply:
I think, This usage without where is the most commonly used style in the project. But I have also seen this kind of for-loop in one place. So, I am not sure If we should change it. Because this code block is also very easy to understand.

Reply:
Uniquing these elements isn't specific to a sequence of `Name`s, so it would be best to extract this to a general purpose `Sequence` extension. That would live in `Sources/ArgumentParser/SequenceExtensions.swift`.